### DISCUSSION TITLE
Alignment issue on multiple page pdf using birt design.

### DISCUSSION THREAD

**Discussion by sharath492** [eclipse-birt/birt#1288] [Q&A]
Hello All,

I need help on one of the pdf alignment issue.

we have xml data with multiple item details so it causing the alignment issue.

Actually i have tested with 100 item details and it's working fine but above 100 items have alignment issues.

So can anyone please help to resolve on this.

I have attached xml data, .rpt design here for your [reference.]([url](url))

Thanks & Regards,
Sharath.
[20230321061913.1114795274194.4.B2BE_Document.pdf](https://github.com/eclipse-birt/birt/files/11540080/20230321061913.1114795274194.4.B2BE_Document.pdf)
[rptDesign_XMLData.zip](https://github.com/eclipse-birt/birt/files/11540122/rptDesign_XMLData.zip)

**Comment by speckyspooky** [comment]
Hello Sharath,

I tried to start your report. I was able to mount your XML-file but which parameters should be set to see a result.
I got an empty report, see my screen.

And what ist the alignment issue, cann you share a screen with the problem.

![2023-05-23 20_24_26-BIRT Report Viewer – Mozilla Firefox](https://github.com/eclipse-birt/birt/assets/41593722/ef24a5f0-316f-407b-84c5-a38f8b2e64b9)

**Reply by sharath492** [reply]
Hello [speckyspooky](https://github.com/speckyspooky),

The parameters values to be set as Y or N to see the result.
LastpageItemNumber = 156
showLastpageNoItem = Y
LastpageNototal = Y
MiddleLastpageWithTotal = Y
FinalReminder = N

Here i shared some screenshots. Please check let me know if anything.

And i have attached one pdf file. can you please look into page No 4,5.

And in the 2nd page the header information is repeated. This should be displayed in only 1st page and footer information should be displayed only on last page but it's displayed in 6,7,8 pages.

![Capture](https://github.com/eclipse-birt/birt/assets/41218746/987d60fc-c04f-4a4f-ae97-99c6351fab13)
![Capture1](https://github.com/eclipse-birt/birt/assets/41218746/a5713059-fa0f-45b8-8c79-e35298744e31)
![Capture2](https://github.com/eclipse-birt/birt/assets/41218746/022224d5-cf8f-48fc-b2c7-87ed76c71b39)

[20230321061913.1114795274194.4.B2BE_Document.pdf](https://github.com/eclipse-birt/birt/files/11551191/20230321061913.1114795274194.4.B2BE_Document.pdf)

**Comment by speckyspooky** [comment]
Your definition is a little bit unclear, because you write about header.
Do you mean the table header or the header based on the pager master.
Your write about footer and I think you mean the text area in your PDF-document.

The table header can be conrolled with the repeat header flag (see above).
If you wants to controll the page amster header info then you woould need a visibility check on the pageNumber-variable to hide rows.
My suggestion would be to set the page-master via "script" on the table and the eseast way would be a 1st-master of 1st-page and 2nd-master of following pages and shift the master page accordingly.

The text of your text area can be controlled via pageNumber == totalPage and the area should be only shown if the compare is true.

But your report isn't clear over all because you use to many tables for the same content.
It yould we easyer to use no so much tables and set the filters or master pages with scripts.

![2023-05-25 21_36_59-workspace - C__Users_gutma_Downloads_rptDesign_XMLData_b2beaccountpdf-aulatepaym](https://github.com/eclipse-birt/birt/assets/41593722/abab17d6-7c04-453a-a589-ec2dc6d1bef0)

**Reply by sharath492** [reply]
Hello [speckyspooky](https://github.com/speckyspooky),

The header and footer information means text area. I have attached screenshots below.
![Capture](https://github.com/eclipse-birt/birt/assets/41218746/aa4cb0c4-694a-4bc8-9694-87ee50a7f820)
![Capture1](https://github.com/eclipse-birt/birt/assets/41218746/91ea8f09-96ad-4526-977a-778f27686aef)

Thank you for suggestion and i'll look into scripts and updated accordingly.

**Comment by speckyspooky** [comment]
Another idea would be to change you master-pages.

If you move the first-page text as a regular text-element on the page layout then it will be printed only 1times.
Yes in this situation you have to redesign your master-page but it will be easier.
The same would be my suggestion for your closing text area. You can add this after your tables
and so it will be printed only 1times.

The master-page is optimal for static text which is repeated on all pages.
To handle there dynamic text like in your case the page-layout with the standard text flow is the better option.
Soif you need the address data only 1times then switch it too (from master- to page-layout).

**Reply by sharath492** [reply]
okay Thank you :). I'll try to change my mastepages.
